addons/web/static/src/js/services/crash_manager.js

```diff
@@ -4,27 +4,128 @@ odoo.define('web.CrashManager', function (require) {
 var ajax = require('web.ajax');
 var core = require('web.core');
 var Dialog = require('web.Dialog');
+var mixins = require('web.mixins');
+var ServicesMixin = require('web.ServicesMixin');
+var ServiceProviderMixin = require('web.ServiceProviderMixin');
+var Widget = require('web.Widget');
 
-var QWeb = core.qweb;
 var _t = core._t;
 var _lt = core._lt;
 
 var map_title ={
-    user_error: _lt('Warning'),
-    warning: _lt('Warning'),
-    access_error: _lt('Access Error'),
-    missing_error: _lt('Missing Record'),
-    validation_error: _lt('Validation Error'),
-    except_orm: _lt('Global Business Error'),
-    access_denied: _lt('Access Denied'),
+    user_error: _lt("User Error"),
+    warning: _lt("Warning"),
+    access_error: _lt("Access Error"),
+    missing_error: _lt("Missing Record"),
+    validation_error: _lt("Validation Error"),
+    except_orm: _lt("Global Business Error"),
+    access_denied: _lt("Access Denied"),
+    karma_error: _lt("Karma Error"),
 };
 
-var CrashManager = core.Class.extend({
+/**
+ * An extension of Dialog Widget to render the warnings and errors on the website.
+ * Extend it with your template of choice like ErrorDialog/WarningDialog
+ */
+var CrashManagerDialog = Dialog.extend({
+    xmlDependencies: (Dialog.prototype.xmlDependencies || []).concat(
+        ['/web/static/src/xml/crash_manager.xml']
+    ),
+
+    /**
+     * @param {Object} error
+     * @param {string} error.message    the message in Warning/Error Dialog
+     * @param {string} error.traceback  the traceback in ErrorDialog
+     *
+     * @constructor
+     */
+    init: function (parent, options, error) {
+        this._super.apply(this, [parent, options]);
+        this.message = error.message;
+        this.traceback = error.traceback;
+    },
+});
+
+var ErrorDialog = CrashManagerDialog.extend({
+    template: 'CrashManager.error',
+});
+
+var WarningDialog = CrashManagerDialog.extend({
+    template: 'CrashManager.warning',
+
+    /**
+     * Sets size to medium by default.
+     *
+     * @override
+     */
+    init: function (parent, options, error) {
+        this._super(parent, _.extend({
+            size: 'medium',
+       }, options), error);
+    },
+
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    /**
+     * Focuses the ok button.
+     *
+     * @override
+     */
+    open: function () {
+        this._super({shouldFocusButtons: true});
+    },
+});
+
+var CrashManager = core.Class.extend(mixins.EventDispatcherMixin, ServicesMixin, ServiceProviderMixin,{
```

Comment:
Here, crash manager and web client are service providers in `/web`.
That means all services are duplicated, each having their own state.
Only the root widget should have `ServiceProviderMixin`, so that each service is deployed once and accessible from any widget with `this.call()` (bubbling up).

To fix this issue, `CrashManager` should extend `AbstractService`.

Reply:
Yes, missed that huge mistake during review. In any case, FJA was working on it but then left for vacations. It will be fixed, don't worry.

Reply:
@qsm-odoo & @alexkuhn I was also bitten at it so once you fix it please refer to https://github.com/odoo/odoo/issues/34910 so this is auto-closed.